### PR TITLE
docs: added stack env variables to new page

### PR DESCRIPTION
**Pull request by Kamran64**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
Created a new page with a table listing all stack environment variables.

## Related Issues
Fixes: https://github.com/appsody/website/issues/122

### CODE REVIEW

File: src/styles/main.css [hunk 1]
```diff
@@ -349,3 +349,22 @@ footer a:hover {
...
+  border: 1px solid lightgray;
+  margin: 0;
+  text-align: left;
+  padding: 6px 12px;
+}
+
+#docs table tr th {
+  border: 1px solid lightgray;
+  margin: 0;
+  column-width: 1200px;
```
Comment (tomleah) [inline src/styles/main.css]: We should avoid using `px` when sizing width and height. With `em` we can size based on font size, I wonder if this would be better?

File: src/styles/main.css [hunk 2]
```diff
@@ -349,3 +349,22 @@ footer a:hover {
 #docs #brand-name {
   display: none;
 }
+
+#docs table tr td {
+  border: 1px solid lightgray;
```
Comment (tomleah) [inline src/styles/main.css]: Try to use the colour palette at the top of `main.css` where possible. Try `var(--light-grey);`

File: src/styles/main.css [hunk 3]
```diff
@@ -349,3 +349,22 @@ footer a:hover {
 #docs #brand-name {
   display: none;
 }
+
+#docs table tr td {
```
Comment (tomleah) [inline src/styles/main.css]: nit: Can we limit these styles further than `#docs`? At the moment this will affect any table on the docs page. 

Try `.doc-content`

**Comment by tomleah** [comment]
Requested review from @kewegner for content

**Comment by tomleah** [comment]
I think we need some context above the table to explain what these environment variables are for. I'm also not 100% on if this page belongs in the sidebar or if it would be a "hidden" doc that is linked to from somewhere else.

**Comment by tomleah** [comment]
@Kamran64 have you had the chance to take a look at this?

**Comment by Kamran64** [comment]
This will be done by this morning - sorry.

File: content/docs/sidebar.yaml [hunk 4]
```diff
@@ -29,4 +29,5 @@
       path: /docs/stacks/stack-structure
 - items:
     - title: FAQ
-      path: /docs/faq
\ No newline at end of file
+      path: /docs/faq
```
Comment (neeraj-laad) [inline content/docs/sidebar.yaml]: why is this file changed?

File: content/docs/stacks/environment-variables.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+---
+path: /docs/stacks/environment-variables
```
Comment (neeraj-laad) [inline content/docs/stacks/environment-variables.md]: we will need a title on this now

File: content/docs/stacks/environment-variables.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+#Environment Variables
```
Comment (neeraj-laad) [inline content/docs/stacks/environment-variables.md]: Appsody environment variables?

File: content/docs/stacks/environment-variables.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+Here you will find a list of environment variables alongside their default values. To use the controller, these Dockerfile environment variables need to be configured. Each stack image exposes environment variables to define commands to be used by Appsody.
```
Comment (neeraj-laad) [inline content/docs/stacks/environment-variables.md]: Stack creators configure Appsody environment variables to specify the behaviour they expect from the stack throughout the application development lifecycle. These variables are defined in `Dockerfile-stack` for the stack image. `Appsody CLI` and `Appsody controller` inspect these environment variables and then drive the expected behaviour for the developer.

If a stack image is built upon another stack's image, it will inherit all the Appsody variables from the base stack. It can override the variables it wants to change. This allows users to create stacks with slightly different behaviour while still getting updates from the base stack.

Following are the currently supported Appsody environment variables and their default values:

File: content/docs/stacks/environment-variables.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
+---
+title: Appsody Environment Variables
+path: /docs/stacks/environment-variables
+---
+#Appsody Environment Variables
```
Comment (tomleah) [inline content/docs/stacks/environment-variables.md]: Please put a space between #'s and the heading
`#Appsody Environment Variables` -> `# Appsody Environment Variables`

Comment (Kamran64) [reply content/docs/stacks/environment-variables.md]: resolved
